pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/ast/GitHubBug208.txt

```diff
@@ -1,22 +1,22 @@
 +- CompilationUnit[@PackageName = ""]
    +- ClassDeclaration[@Abstract = false, @Annotation = false, @Anonymous = false, @BinaryName = "GitHubBug208", @CanonicalName = "GitHubBug208", @EffectiveVisibility = Visibility.V_PUBLIC, @Enum = false, @Final = false, @Interface = false, @Local = false, @Nested = false, @PackageName = "", @Record = false, @RegularClass = true, @RegularInterface = false, @SimpleName = "GitHubBug208", @Static = false, @TopLevel = true, @Visibility = Visibility.V_PUBLIC]
-      +- ModifierList[@EffectiveModifiers = "{public}", @ExplicitModifiers = "{public}"]
+      +- ModifierList[@EffectiveModifiers = (JModifier.PUBLIC), @ExplicitModifiers = (JModifier.PUBLIC)]
```

Comment:
Do these changes affect XPath rules? Do we need to anything here regarding compatibility?

Reply:
Hm... we have this custom XPath function https://docs.pmd-code.org/latest/pmd_userdocs_extending_writing_xpath_rules.html#pmd-java-explicitModifiers
Maybe this is not needed anymore now and can be deprecated?

Reply:
Yes, I agree, but I would rather do that in a separate PR… we should deprecate it, place mechanisms to log it's usage (as we do for deprecated attributes), and replace all usages within our own rules… that's probably going to be a chunky changeset, and this PR is already at ~1200 changed LoCs…

Reply:
Yes, let's do this in a separate PR.